practical/groovy/analyse_image_map_annotation.groovy

```diff
@@ -0,0 +1,157 @@
+ *  (at your option) any later version.
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * ------------------------------------------------------------------------------
+ */
+
+/*
+ * This Groovy script uses ImageJ to crop an image.
+ * The cropped image is saved locally as OME-TIFF using Bio-Formats Exporter.
+ * Upload the OME-TIFF to OMERO.
+ * Use this script in the Scripting Dialog of Fiji (File > New > Script).
+ * Select Groovy as language in the Scripting Dialog.
+ * Error handling is omitted to ease the reading of the script but this
+ * should be added if used in production to make sure the services are closed
+ * Information can be found at
+ * https://docs.openmicroscopy.org/latest/omero5/developers/Java.html
+ */
+
+import java.util.ArrayList
+import java.util.List
+import java.io.File
+import java.lang.reflect.Array
+
+// OMERO Dependencies
+import omero.model.NamedValue
+import omero.model.ImageI
+import omero.gateway.Gateway
+import omero.gateway.LoginCredentials
+import omero.gateway.SecurityContext
+import omero.gateway.facility.BrowseFacility
+import omero.gateway.facility.DataManagerFacility
+import omero.gateway.model.ImageData
+import omero.gateway.model.MapAnnotationData
+import omero.log.SimpleLogger
+
+import ome.formats.importer.ImportConfig
+import ome.formats.importer.OMEROWrapper
+import ome.formats.importer.ImportLibrary
+import ome.formats.importer.ImportCandidates
+import ome.formats.importer.cli.ErrorHandler
+import ome.formats.importer.cli.LoggingImportMonitor
+
+import loci.formats.in.DefaultMetadataOptions
+import loci.formats.in.MetadataLevel
+
+import ij.IJ
+import ij.plugin.frame.RoiManager
+import ij.measure.ResultsTable
+
+// Setup
+// =====
+
+// OMERO Server details
+HOST = "outreach.openmicroscopy.org"
+PORT = 4064
+group_id = "-1"
+// parameters to edit
+image_id = 1001
+USERNAME = "username"
+PASSWORD = "password"
+
+def connect_to_omero() {
+    "Connect to OMERO"
+
+    credentials = new LoginCredentials()
+    credentials.getServer().setHostname(HOST)
+    credentials.getServer().setPort(PORT)
+    credentials.getUser().setUsername(USERNAME.trim())
+    credentials.getUser().setPassword(PASSWORD.trim())
+    simpleLogger = new SimpleLogger()
+    gateway = new Gateway(simpleLogger)
+    gateway.connect(credentials)
+    return gateway
+
+}
+
+def open_image_plus(HOST, USERNAME, PASSWORD, PORT, group_id, image_id) {
+    "Open the image using the Bio-Formats Importer"
+
+    StringBuffer options = new StringBuffer()
```

Comment:
Use `StringBuilder` if you don't need thread-safety.

Reply:
I will keep it as it is for now since I do not have time to review all the other scripts.

Reply:
Okay.

Reply:
Saying that I will fix those 2 and open another PR
the power of cut & paste :-)